calico/felix/devices.py

```diff
@@ -95,6 +97,49 @@ def interface_exists(interface):
 
 def list_interface_ips(ip_type, interface):
     """
+    List the local IPs assigned to an interface.
+    :param str ip_type: IP type, either futils.IPV4 or futils.IPV6
+    :param str interface: Interface name
+    :returns: a set of all addresses directly assigned to the device.
+    """
+    if ip_type == futils.IPV4:
+        data = futils.check_call(
+            ["ip", "addr", "list", "dev", interface]).stdout
+        regex = r'^    inet ([0-9.]+)'
+    else:
+        data = futils.check_call(
+            ["ip", "-6", "addr", "list", "dev", interface]).stdout
+        regex = r'^    inet6 ([0-9:]+)'
+    # Search the output for lines beginning "    inet(6)".
+    ips = re.findall(regex, data, re.MULTILINE)
+    _log.debug("Interface %s has %s IPs %s", interface, ip_type, ips)
+    return set([IPAddress(ip) for ip in ips])
```

Comment:
Would it be slightly neater to skip creating the list, and just use a generator comprehension?

```python
return set(IPAddress(ip) for ip in ips)
```